### PR TITLE
Improvements to Window Manipulation Transient state

### PR DESCRIPTION
**Pull request by yuhan0**
- Adds `SPC w [`, `SPC w ]`, `SPC w {`, `SPC w }` leader key bindings for easier entry into transient state resizing

- Allow window transient state to be collapsed via "?" for more accurate presentation of layouts during the TS.

- Add `m`, `|` and `_` bindings to the transient state for consistency with the `SPC w` bindings

### CODE REVIEW

**Comment by yuhan0** [comment]
Here's a screenshot showing the changes in expanded state:
expanded
Minifed state:
minified

File: layers/+spacemacs/spacemacs-defaults/keybindings.el [hunk 1]
```diff
@@ -628,22 +628,45 @@ respond to this toggle."
...
- [_o_]^^    other frame   ^^^^                  ^^                    ^^                   "
-               (if (configuration-layer/package-used-p 'golden-ratio)
-                   "[_g_] golden-ratio %`golden-ratio-mode"
-                 "")
-               "\n ^^^^                     ^^^^                  ^^                    ^^                   [_q_] quit")
+  :hint-is-doc t
+  :dynamic-hint (spacemacs//window-manipulation-ts-hint)
   :bindings
   ("q" nil :exit t)
+  ("?" spacemacs//window-manipulation-ts-toggle-hint)
```
Comment (sdwolfz) [inline layers/+spacemacs/spacemacs-defaults/keybindings.el]: 👏

**Comment by duianto** [comment]
`[?] help` isn't listed in the expanded state.

Is it enough to only show it in the minified state since it opens first and one has to press `?` to show the help? Or should all the transient state keys be listed?

Another solution could be to add it to the right of the title, and since the help popup already is open, maybe it could say something like this:
`Window Manipulation Transient State ([?] hide help)`
or
`Window Manipulation Transient State ([?] toggle help)`

**Comment by duianto** [comment]
Some commands abbreviate transient state as `-ts-`, so maybe `Window Manipulation TS`, or if transient state doesn't need to be mentioned, Then it could just say `Window Manipulation`.

**Comment by sdwolfz** [comment]
Can we have "TS" for the minified and "Transient State" for the other? if not "TS" for both should be enough.

**Comment by duianto** [comment]
The parentheses around `([?] help)` might not be needed either, that would save 2 chars 😄

**Comment by duianto** [comment]
The help key could be shown the same way as the other keys in the minified state:
```
Window Manipulation TS  Help: ? Select: w h ...
```

**Comment by yuhan0** [comment]
Yeah I personally feel that "Transient State" is a little redundant, but it's in line with all the other transient state titles. It doesn't seem like the hydra titles can be changed dynamically, although there's surely a hack around it. 

@duianto I took the formatting from the current Layout and Workspaces transient states where they also use parens around `([?] help)`.

**Comment by yuhan0** [comment]
Does this look alright for the expanded state?
image

**Comment by duianto** [comment]
This can be the new template, and we'll fix the others based on this. They would probably also benefit from having a minified state.

**Comment by yuhan0** [comment]
Layouts and Workspaces are the ones which *do* currently have a minified state, and I adapted most of the code structure from them. No strong opinions either way, up to the maintainers to make a call :)

> @sdwolfz 
> Can we have "TS" for the minified and "Transient State" for the other?

I figured out an ugly hack to change the title:
```emacs-lisp
  (setf (cadr spacemacs/window-manipulation-transient-state/hint)
        (concat (substring (cadr spacemacs/window-manipulation-transient-state/hint) 0 20)
                (propertize (if (zerop spacemacs--ts-full-hint-toggle) "Transient State" "TS")
                            'face 'spacemacs-transient-state-title-face)))
```

This can be stuck inside `spacemacs//window-manipulation-ts-toggle-hint` or the :on-enter prop of the TS, but I don't think messing with the hydra's internal representation is such a good idea.

**Comment by sdwolfz** [comment]
I don't really think other TS would benefit from this. Window manipulation fits perfectly since you see exactly what the end result will be in real time.

@yuhan0 😄 let's skip it if it's a hack and go with `TS` for both.

**Comment by yuhan0** [comment]
Alright, just pushed the changes :)

**Comment by duianto** [comment]
> I don't really think other TS would benefit from this

👍

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.

**Pull request closed without merging**